jwst/pipeline/calwebb_coron3.py

```diff
@@ -46,17 +46,25 @@ class Coron3Pipeline(Pipeline):
 
     prefetch_references = False
 
+    # Main processing
     def process(self, user_input):
-        """Primary method for performing pipeline."""
+        """Primary method for performing pipeline.
+
+        Parameters
+        ----------
+        user_input: str, Level3 Association, or ~jwst.datamodels.DataModel
+            The exposure or association of exposures to process
+        """
         self.log.info('Starting calwebb_coron3 ...')
+        asn_exptypes = ['science', 'psf']
 
-        # Load the input association table
-        asn = self.load_as_level3_asn(user_input)
-        acid = asn.get('asn_id', '')
+        # Create a DM object using the association table
+        input_models = datamodels.open(user_input, asn_exptypes=asn_exptypes)
+        acid = input_models.meta.asn_table.asn_id
 
         # We assume there's one final product defined by the association
-        prod = asn['products'][0]
-        self.output_file = prod.get('name', self.output_file)
+        prod = input_models.meta.asn_table.products[0].instance
+        self.output_file = input_models.meta.asn_table.products[0].name
```

Comment:
There is :
        input_models.meta.asn_table.products[0].name

   but not :
       input_models.meta.asn_table.products[0].instance.name

    . . .  so we can't get `name` directly from `prod` that way, since
     prod = input_models.meta.asn_table.products[0].instance

Reply:
Hmm, `asn_table` should be the association structure. So, the whole specification to the name should be
```
prod = input_models.meta.asn_table.products[0]
self.output_file = prod['name']
```

As mentioned previously, `ModelContainer.meta.asn_table` is just the association `dict` structure, as described [in the docs](https://jwst-pipeline.readthedocs.io/en/latest/jwst/associations/level3_asn_technical.html#logical-structure).

@hbushouse mentions that the `instance` is also done in Spec3Pipeline, which should also be re-examined. Looking at `ModelContainer`, `asn_table` is just the `dict` structure.